src/java/transform/visitors/TransformVisitor.java

```diff
@@ -575,75 +578,109 @@ public boolean visit(MethodInvocation node) {
-				replaceInteger(node);
-			} else if (isBooleanTypeCode(type)) {
-				replaceBoolean(node);
-			} else if (isFloatingPointTypeCode(type)) {
-				replaceFloat(node);
-			} else if (isDoubleTypeCode(type)) {
-				replaceDouble(node);
-			} else {
-				if (node.getParent().getParent() instanceof Block) {
-					rewriter.remove(node.getParent().getParent(), null);
-				} else {
-					rewriter.replace(node.getParent().getParent(), ast.newBlock(), null);
-				}
-				typeTable.setNodeType(lhs, null);
-			}
-
-		} else if (node.getLocationInParent() == VariableDeclarationFragment.INITIALIZER_PROPERTY) {
-			VariableDeclarationFragment parent = (VariableDeclarationFragment) node.getParent();
-			Type type = typeTable.getNodeType(parent);
-
-			if (isIntegerTypeCode(type)) {
-				replaceInteger(node);
-				return;
-			} else if (isBooleanTypeCode(type)) {
-				replaceBoolean(node);
-				return;
-			} else if (isFloatingPointTypeCode(type)) {
-				replaceFloat(node);
-				return;
-			} else if (isDoubleTypeCode(type)) {
-				replaceDouble(node);
-				return;
-			} else {
-				rewriter.remove(node, null);
-				typeTable.setNodeType(parent, null);
-			}
-			
-		} else if (node.getLocationInParent() == CastExpression.EXPRESSION_PROPERTY) {
-			CastExpression parent = (CastExpression) node.getParent();
-			Type type = parent.getType();
-
-			// if the type directly above is a cast, we can ignore the need for it as it is implicit in the new symbolic value
-			if (isIntegerTypeCode(type)) {
-				replaceInteger(parent);
-				typeTable.setNodeType(parent.getParent(), ast.newPrimitiveType(PrimitiveType.INT));
-			} else if (isBooleanTypeCode(type)) {
-				replaceBoolean(parent);
-				typeTable.setNodeType(parent.getParent(), ast.newPrimitiveType(PrimitiveType.BOOLEAN));
-			} else if (isFloatingPointTypeCode(type)) {
-				replaceDouble(node);
-				typeTable.setNodeType(parent, ast.newPrimitiveType(PrimitiveType.FLOAT));
-			} else if (isDoubleTypeCode(type)) {
-				replaceDouble(parent);
-				typeTable.setNodeType(parent.getParent(), ast.newPrimitiveType(PrimitiveType.DOUBLE));
-			}
-		} 
+		IMethodBinding methodBinding = node.resolveMethodBinding();
+        if (methodBinding != null) {
+            ITypeBinding declaringClass = methodBinding.getDeclaringClass();
+            if (declaringClass != null) {
+                String packageName = declaringClass.getPackage().getName();
+                if (rootNodePackage == null) {
+                	rootNodePackage = ((CompilationUnit) node.getRoot()).getPackage().getName().getFullyQualifiedName();
+                }
+                // Check if it's part of the JDK
+                if (packageName.startsWith("java.") || packageName.startsWith("javax.") /* || packageName.equals(rootNodePackage) */) {
+                	// TODO: clean up this dead code and add checks for if package starts with package project is contained in
+                	// TODO: make this play nice with cast expressions
```

Comment:
Was this TODO completed?

Reply:
It should be working correctly with cast expressions; will write more tests soon!